src/libraries/System.Net.Sockets/tests/FunctionalTests/NetworkStreamTest.cs

```diff
@@ -317,14 +317,14 @@ public async Task DisposeSocketDirectly_ReadWriteThrowIOException()
                 {
                     serverSocket.Dispose();
 
-                    Assert.Throws<IOException>(() => server.Read(new byte[1], 0, 1));
-                    Assert.Throws<IOException>(() => server.Write(new byte[1], 0, 1));
+                    Assert.Throws<NetworkException>(() => server.Read(new byte[1], 0, 1));
+                    Assert.Throws<NetworkException>(() => server.Write(new byte[1], 0, 1));
 
-                    Assert.Throws<IOException>(() => server.BeginRead(new byte[1], 0, 1, null, null));
-                    Assert.Throws<IOException>(() => server.BeginWrite(new byte[1], 0, 1, null, null));
+                    Assert.Throws<NetworkException>(() => server.BeginRead(new byte[1], 0, 1, null, null));
+                    Assert.Throws<NetworkException>(() => server.BeginWrite(new byte[1], 0, 1, null, null));
 
-                    Assert.Throws<IOException>(() => { server.ReadAsync(new byte[1], 0, 1); });
-                    Assert.Throws<IOException>(() => { server.WriteAsync(new byte[1], 0, 1); });
+                    Assert.Throws<NetworkException>(() => { server.ReadAsync(new byte[1], 0, 1); });
+                    Assert.Throws<NetworkException>(() => { server.WriteAsync(new byte[1], 0, 1); });
```

Comment:
I know it's a deficiency of the original tests, but shouldn't we add validation for `exception.NetworkError` with this refactor?

Reply:
We could, but these are just wrapping ObjectDisposedException so the NetworkError would be Unknown, which isn't all that useful to validate.